src/client/pages/Offer/ProductSelector/ProductCard.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {
+  useRef,
+  forwardRef,
+  ComponentPropsWithoutRef,
+  MouseEventHandler,
+} from 'react'
+import styled from '@emotion/styled'
+import { mergeRefs } from 'react-merge-refs'
+import { colorsV3 } from '@hedviginsurance/brand'
+import { Switch as Checkbox } from 'components/Switch'
+import { MEDIA_QUERIES } from 'utils/mediaQueries'
+
+export type Product = {
+  id: string
+  name: string
+  description: string
+  price: string
+}
+
+type ProductCardProps = {
+  product: Product
+} & Required<Pick<ComponentPropsWithoutRef<'input'>, 'checked' | 'onChange'>>
+
+export const ProductCard = forwardRef<HTMLInputElement, ProductCardProps>(
+  ({ product, checked, onChange }, forwardedRef) => {
+    const checkboxRef = useRef<HTMLInputElement | null>(null)
+
+    const handleClick: MouseEventHandler<HTMLLIElement> = () => {
+      checkboxRef.current?.click()
+    }
+
+    return (
+      <Card selected={checked} onClick={handleClick}>
+        <header>
+          <Heading>{product.name}</Heading>
+          <Subheading>{product.price}</Subheading>
+        </header>
+        <Description>{product.description}</Description>
+        <CheckboxWrapper>
+          <Checkbox
+            ref={mergeRefs([checkboxRef, forwardedRef])}
```

Comment:
Added `react-merge-refs` here because it should be possible to pass in a `ref` to `ProductCard` which will forward it to its checkbox. However `ProductCard` itself also need to hold a `ref` to its checked because of the _click_ mechanism. Therefore the requirement of merging the refs.